iocore/cache/CacheWrite.cc

```diff
@@ -106,7 +106,13 @@ CacheVC::updateVector(int /* event ATS_UNUSED */, Event * /* e ATS_UNUSED */)
          data should remain valid.
       */
       if (alternate_index >= 0) {
-        alternate.copy_frag_offsets_from(write_vector->get(alternate_index));
+        /*if length changed, fragment offset also need to change, if not cause range request error*/
+        if (total_len != write_vector->get(alternate_index)->object_size_get()) {
```

Comment:
Sometime we were try to update the header only fragment that means we won't write any fragment in it. So the `total_len` is always 0 in this situation.  And we will lost the fragments info forever if we don't copy the fragments table from old info.

Reply:
There is situation that first response code is 404 and next response code is 200,so the total_len has changed to non zero.In this situation,this cause data read error.

Reply:
But it affect the normal requests.

Reply:
You can test this situation.